demos/https/iot_demo_https_s3_download_async.c

```diff
@@ -519,12 +523,12 @@ static bool _getNextIncompleteRange( uint32_t * currentRange )
     bool found = false;
     /* An intermediate bitMask calculation for checking if the currentRange is already downloaded or not. */
     uint32_t bitMask = 0;
-    /* An intermediate byteOffset into the downlaodedBitmap to check if the currentRange is already downloaded or not. */
+    /* An intermediate byteOffset into the downloadedBitmap to check if the currentRange is already downloaded or not. */
     uint32_t byteOffset = 0;
 
     /* A count from 0 to check if the total ranges is reached because, in the logic below, currentRange will wrap
      * around to check the start of the bitmap. */
-    int32_t rangeCheckCount = 0;
+    uint32_t rangeCheckCount = 0;
 
     do
     {
```

Comment:
Can you please write a comment on what the first several lines of the below block are doing?

Reply:
```
        /* Check if the current range in the file is already downloaded or is has already been scheduled. The 
         * bitmaps contains as many bits are there are blocks of byte ranges. The lines below help to index 
         * into the bitmaps and check if the bit representing that byte range is set. */
        bitMask = BITMASK( *currentRange );
        byteOffset = BYTE_OFFSET( *currentRange );

        /* If this range has not been downloaded yet and has not been schedule, then return the current range. */
        if( ( ( _fileDownloadInfo.downloadedBitmap[ byteOffset ] & bitMask ) == ( ( uint8_t ) 0 ) ) &&
            ( ( _fileDownloadInfo.scheduledBitmap[ byteOffset ] & bitMask ) == ( ( uint8_t ) 0 ) ) )
        {
            break;
        }
```

Reply:
"bitmaps contains"

Reply:
thanks